### PR TITLE
Eslint fixes: max-len

### PR DESCRIPTION
**Pull request by reosarevok**
https://eslint.org/docs/rules/max-len

Skips one case that is being removed with https://github.com/metabrainz/musicbrainz-server/pull/1698

### CODE REVIEW

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -456,7 +459,13 @@ const CLEANUPS = {
     },
   },
   'amazon': {
-    match: [new RegExp('^(https?://)?(((?!music)[^/])+\.)?(amazon\\.(com|ca|co\\.uk|fr|ae|at|de|it|sg|co\\.jp|jp|cn|es|in|nl|com\\.br|com\\.mx|com\\.au|com\\.tr)|amzn\\.com)', 'i')],
+    match: [new RegExp(
+      '^(https?://)?(((?!music)[^/])+\.)?(amazon\\.' +
+      '(com|ca|co\\.uk|fr|ae|at|de|it|sg|co\\.jp|jp|' +
+      'cn|es|in|nl|com\\.br|com\\.mx|com\\.au|com\\.tr)' +
+      '|amzn\\.com)',
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js:466]: This improves readability by a lot! :+1: I would even suggest to further split by independent part, to use parenthesis for indent, and to sort domain names from the left most part (but that one should probably be a separate commit).
```suggestion
      '^(https?://)?' +
      '(((?!music)[^/])+\.)?' +
      '(amazon\\.(' + (
        'ae|at|com\\.au|com\\.br|ca|cn|com|de|es|fr|in' +
        '|it|jp|co\\.jp|com\\.mx|nl|sg|com\\.tr|co\\.uk'
      ) + ')|amzn\\.com)',
```

File: root/static/scripts/edit/URLCleanup.js [hunk 2]
```diff
@@ -493,7 +502,13 @@ const CLEANUPS = {
     },
   },
   'amazonmusic': {
-    match: [new RegExp('^(https?://)?music\\.amazon\\.(com|ca|co\\.uk|fr|ae|at|de|it|sg|co\\.jp|jp|cn|es|in|nl|com\\.br|com\\.mx|com\\.au|com\\.tr)/(albums|artists)', 'i')],
+    match: [new RegExp(
+      '^(https?://)?music\\.amazon\\.' +
+      '(com|ca|co\\.uk|fr|ae|at|de|it|sg|co\\.jp|jp|' +
+      'cn|es|in|nl|com\\.br|com\\.mx|com\\.au|com\\.tr)' +
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js:508]: Same as above:

```suggestion
      '(ae|at|com\\.au|com\\.br|ca|cn|com|de|es|fr|in' +
      '|it|jp|co\\.jp|com\\.mx|nl|sg|com\\.tr|co\\.uk)' +
```

File: root/static/scripts/relationship-editor/common/fields.js [hunk 3]
```diff
@@ -538,8 +543,10 @@ class Relationship {
       `&conditions.1.name=${encodeURIComponent(entity1.name)}` +
       `&conditions.1.args.0=${encodeURIComponent(entity1.id)}` +
       '&conditions.2.field=type' +
-      '&conditions.2.operator=%3D&conditions.2.args=90%2C233&conditions.2.args=91' +
-      '&conditions.2.args=92&conditions.3.field=status&conditions.3.operator=%3D' +
+      '&conditions.2.operator=%3D' +
+      '&conditions.2.args=90%2C233&conditions.2.args=91' +
+      '&conditions.2.args=92' +
+      '&conditions.3.field=status&conditions.3.operator=%3D' +
       '&conditions.3.args=1&field=Please+choose+a+condition'
```
Comment (yvanzo) [inline root/static/scripts/relationship-editor/common/fields.js:550]:
```suggestion
      '&conditions.2.args=90%2C233' +
      '&conditions.2.args=91' +
      '&conditions.2.args=92' +
      '&conditions.3.field=status' +
      '&conditions.3.operator=%3D' +
      '&conditions.3.args=1' +
      '&field=Please+choose+a+condition'
```

File: root/static/scripts/release-editor/fields.js [hunk 4]
```diff
@@ -961,7 +1008,9 @@ class Release extends mbEntity.Release {
     this.needsTrackInfo = errorField(function () {
       return !self.hasTrackInfo();
     });
-    this.hasInvalidPregapLength = errorField(this.mediums.any('hasInvalidPregapLength'));
+    this.hasInvalidPregapLength = errorField(
+      this.mediums.any('hasInvalidPregapLength'),
+    );
```
Comment (yvanzo) [inline root/static/scripts/release-editor/fields.js:1013]: I would suggest to simply make eslint to ignore the files that need to be replaced because they are based on Knockout and jQuery rather than React.

**Review by yvanzo — CHANGES_REQUESTED** [review]
It should be rebased and move `max-len` from `.eslintrc.unfixed.yaml` to `.eslintrc.yaml`.

File: root/report/SuperfluousDataTracks.js [hunk 5]
```diff
@@ -38,7 +42,8 @@ const SuperfluousDataTracks = ({
            {data_track_guidelines|data track guidelines}.`,
           {
             data_track_guidelines:
-            `/doc/Style/Unknown_and_untitled/Special_purpose_track_title#Data_tracks`,
+              '/doc/Style/Unknown_and_untitled/' +
+              'Special_purpose_track_title#Data_tracks',
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/report/SuperfluousDataTracks.js:46]: It already allows to avoid splitting URLs, but this isn't recognized as one, IIRC. I don't think this change hurts tbh.

File: root/static/scripts/release-editor/fields.js (hunk 4 above)
Comment (reosarevok) [reply root/static/scripts/release-editor/fields.js:1034]: I don't think we want it to fully ignore these - if it did, we also wouldn't find out if someone added a more serious issue that eslint would have caught :) (and we won't get rid of most of them that quickly either).

File: root/static/scripts/tests/release-editor/actions.js [hunk 6]
```diff
@@ -42,27 +42,55 @@ test('removing a medium should change the medium positions', function (t) {
...
-  t.equal(tracks[0], originalTrack2, 'original track 2 has moved to position 1');
-  t.equal(originalTrack2.position(), 1, 'original track 2 now has position() 1');
-
-  t.equal(tracks[1], originalTrack1, 'original track 1 has moved to position 2');
-  t.equal(originalTrack1.position(), 2, 'original track 1 now has position() 2');
-});
+test(
+  `reordering tracks that have non-consecutive
+   "position" properties (MBS-7227)`,
+  function (t) {
```
Comment (mwiencek) [inline root/static/scripts/tests/release-editor/actions.js:62]: This is unfortunate, since we had to rewrite all these files for the indent fixes PR. :( They were using four spaces before, and now we're changing them all back to that...

I think there is a way to avoid this by wrapping the first argument in parentheses, which I'd much prefer:

```JS
test((
  `reordering tracks that have non-consecutive
   "position" properties (MBS-7227)`
), function (t) {
  // this only requires two spaces now
});
```

Comment (reosarevok) [reply root/static/scripts/tests/release-editor/actions.js]: Made that change (which also thankfully made this PR a lot smaller)

File: root/static/scripts/edit/URLCleanup.js [hunk 7]
```diff
@@ -892,9 +916,12 @@ const CLEANUPS = {
       if (m) {
         const frBnF = m[1];
         const phbt = '0123456789bcdfghjkmnpqrstvwxz';
-        const controlChar = phbt[Array.from(frBnF).reduce(function (sum, c, i) {
```
Comment (mwiencek) [inline root/static/scripts/edit/URLCleanup.js:895]: Removing `function ` here (using an arrow function) would more simply resolve the max-len issue.

File: root/static/scripts/edit/confirmNavigationFallback.js [hunk 8]
```diff
@@ -35,9 +35,13 @@ MB.confirmNavigationFallback = function () {
       const node = event.srcElement || event.target;
       const tag = node.tagName.toLowerCase();
       const type = (node.type || '').toLowerCase();
-      const prevent = !((tag == 'input' && (type == 'text' || type == 'password')) || tag == 'textarea');
+      const prevent = !(
+        (tag == 'input' && (type == 'text' || type == 'password')) ||
+        tag == 'textarea');
```
Comment (mwiencek) [inline root/static/scripts/edit/confirmNavigationFallback.js:40]: I don't know if this violates any rule but I'd keep parentheses consistent with `brace-style`
```suggestion
        tag == 'textarea'
      );
```

File: root/static/scripts/tests/Control/URLCleanup.js [hunk 9]
```diff
@@ -4135,46 +4140,82 @@ function doMatchSubtest(
...
+          subtest.input_url,
+          'input',
+          subtest.expected_relationship_type,
+        );
         tested = true;
       } else {
-        st.fail('Test is invalid: "input_entity_type" is specified without "expected_relationship_type".');
+        st.fail(
+          `Test is invalid: "input_entity_type" is specified
+           without "expected_relationship_type".`,
```
Comment (mwiencek) [inline root/static/scripts/tests/Control/URLCleanup.js:4183]: Using template strings here is going to add all of the whitespace and newlines to the output. Same for all the other test files or non-translated strings.

File: t/selenium.js [hunk 10]
```diff
@@ -614,39 +627,43 @@ async function runCommands(commands, t) {
     const isLastTest = index === testsToRun.length - 1;
 
     return new Promise(function (resolve) {
-      test(title, {objectPrintDepth: 10, timeout: TEST_TIMEOUT}, function (t) {
```
Comment (mwiencek) [inline t/selenium.js:617]: I'd just do `const testOptions = {objectPrintDepth: 10, timeout: TEST_TIMEOUT};`

File: webpack.client.config.js [hunk 11]
```diff
@@ -107,14 +108,21 @@ function createJsPo(srcPo, lang) {
    * possible terminal paths.
    */
   const scriptsDir = shellQuote.quote([dirs.SCRIPTS]);
-  const nestedDirs = shell.exec(`find ${scriptsDir} -type d`, {silent: true}).stdout.split('\n');
+  const nestedDirs = shell.exec(
+    `find ${scriptsDir} -type d`,
+    {silent: true}
```
Comment (mwiencek) [inline webpack.client.config.js:113]:
```suggestion
    {silent: true},
```

File: webpack.client.config.js [hunk 12]
```diff
@@ -107,14 +108,21 @@ function createJsPo(srcPo, lang) {
...
+  ).stdout.split('\n');
   const msgLocations = nestedDirs
     .filter(Boolean)
-    .map(dir => '-N ' + shellQuote.quote(['..' + dir.replace(dirs.CHECKOUT, '') + '/*.js']))
+    .map(dir => (
+      '-N ' +
+      shellQuote.quote(['..' + dir.replace(dirs.CHECKOUT, '') + '/*.js'])))
```
Comment (mwiencek) [inline webpack.client.config.js:119]:
```suggestion
      shellQuote.quote(['..' + dir.replace(dirs.CHECKOUT, '') + '/*.js']))
    )
```

Comment (reosarevok) [reply webpack.client.config.js:119]: This one gets hit with `Unexpected newline before ')'.eslintfunction-paren-newline
`

File: root/static/scripts/edit/forms.js [hunk 13]
```diff
@@ -121,7 +121,9 @@ ko.bindingHandlers.loop = {
     const elements = options.elements || {};
     const template = [];
 
-    for (const node of Array.from(ko.virtualElements.childNodes(parentNode))) {
+    for (const node of Array.from(
+      ko.virtualElements.childNodes(parentNode),
+    )) {
```
Comment (mwiencek) [inline root/static/scripts/edit/forms.js:126]:
```suggestion
    const childNodes = Array.from(ko.virtualElements.childNodes(parentNode));
    for (const node of childNodes) {
```

File: root/static/scripts/release-editor/trackParser.js [hunk 14]
```diff
@@ -134,15 +134,16 @@ const trackParser = releaseEditor.trackParser = {
...
+    sortByNumber(dataTrackPairs, x => -x.similarity)
+      .forEach(function (match) {
+        var data = match.data;
+        var track = match.track;
+
+        if (!data.matchedTrack && !matchedTracks[track.uniqueID]) {
+          data.matchedTrack = track;
+          matchedTracks[track.uniqueID] = 1;
+        }
+      });
```
Comment (mwiencek) [inline root/static/scripts/release-editor/trackParser.js:146]: To make the diff smaller here I'd revert and just change `forEach(function (match) {` to `forEach((match) => {`

**Review by mwiencek — CHANGES_REQUESTED** [review]

File: root/static/scripts/relationship-editor/common/fields.js [hunk 15]
```diff
@@ -500,13 +501,17 @@ class Relationship {
   }
 
   showLinkOrder(source) {
-    return this.linkOrder() > 0 && this.entityCanBeReordered(this.target(source));
+    return this.linkOrder() > 0 &&
+      this.entityCanBeReordered(this.target(source));
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/static/scripts/relationship-editor/common/fields.js:505]: Seems like most by far use the style you suggest here, so sticking to that (changed the couple others I noticed)

**Comment by reosarevok** [comment]
Added a separate .eslintrc for root/static/scripts/tests to ignoreStrings for max-len

File: root/static/scripts/tests/Control/URLCleanup.js [hunk 16]
```diff
@@ -4136,46 +4141,82 @@ function doMatchSubtest(
...
+        );
         st.end();
         return;
       }
     } else if ('expected_relationship_type' in subtest) {
-      st.fail('Test is invalid: "expected_relationship_type" is specified without "input_entity_type".');
+      st.fail(
+        'Test is invalid: "expected_relationship_type" is specified ' +
+        'without "input_entity_type".',
+      );
```
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js:4193]: These strings should not be split, the new `tests/.eslintrc` probably allows that.

File: root/static/scripts/tests/Control/URLCleanup.js [hunk 17]
```diff
@@ -4136,46 +4141,82 @@ function doMatchSubtest(
...
+        );
       }
       tested = true;
     }
     if (subtest.input_relationship_type && !subtest.only_valid_entity_types) {
-      st.fail('Test is invalid: "input_relationship_type" is specified without "only_valid_entity_types" array.');
+      st.fail(
+        'Test is invalid: "input_relationship_type" is specified ' +
+        'without "only_valid_entity_types" array.',
+      );
```
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js:4219]: Same as above.

File: root/static/scripts/tests/Control/URLCleanup.js [hunk 18]
```diff
@@ -4184,26 +4225,37 @@ testData.forEach(function (subtest, i) {
         subtest.expected_relationship_type;
       const cleanUrl = subtest.expected_clean_url || actualCleanUrl;
       if (!relationshipType) {
-        st.fail('Test is invalid: "only_valid_entity_types" are specified with neither "expected_relationship_type" nor "input_relationship_type".');
+        st.fail(
+          'Test is invalid: "only_valid_entity_types" are specified ' +
+          'with neither "expected_relationship_type" ' +
+          'nor "input_relationship_type".',
+        );
```
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js:4232]: Same as above.

**Review by yvanzo — CHANGES_REQUESTED** [review]
Thanks, that looks good overall. Just noticed a few still split strings under `tests/`.

**Review by yvanzo — APPROVED** [review]
Thanks, this patch improves readability in many cases (e.g. too long list of properties in component instantiation, too long expressions with a ternary operator…) even though there are some lesser frequent cases where it doesn’t imho (e.g. splitting string value such as a path, splitting line that ends with a string, splitting test data item…).
As discussed during the latest MB dev meeting, please don’t make such changes again to files that must be rewritten anyway (e.g. files requiring Knockout or jQuery).

**Review by mwiencek — APPROVED** [review]
